### ISSUE TITLE
User shall be able to specify "dynamic" parameters for the pipelines

### ISSUE THREAD

**Issue by sidoruka** [epam/cloud-pipeline#1215]
**Background**
At the moment, Cloud Pipeline does not allow to specify **dynamic** parameters for pipeline configuration, e.g. show/hide some parameters if another parameter has specific value. It would be nice to implement **visibility** feature for pipeline configuration parameters

**Approach**
Extend pipeline's **config.json** to support the following structures:
- **visible** field for parameter:
```
...
parameters: {
  "p1": {
     "type" : "enum",
     "enum" : ["human", "mouse"],
     "required" : true
  },
  "p2": {
     "type" : "string",
     "visible": "p1 == 'human'"
  }
}
...
```
`"visible": "p1 == 'human'"` means **p2** parameter will be hidden if user specifies *human* value for **p1** parameter;
- **visible** field for enumeration *values*:
```
...
parameters: {
  "p1": {
     "type" : "enum",
     "enum" : ["human", "mouse"],
     "required" : true
  },
  ...
  "p3": {
     "type" : "enum",
     "enum": [
        {
           "name": "hg19",
            "visible": "p1 == 'human'"
        },
        {
           "name": "hg38",
            "visible": "p1 == 'human'"
        }
        ...
     ],
  }
}
...
```
**p3** parameter will have *hg19* and *hg38* options available only if **p1** parameter is set to *human*

**Visibility configuration**
To configure parameter's or value's visibility, user must provide an expression that results to **true** or **false**. User can use parameter **names**, strings (in single/double quotas), numbers and *&&* (logical "and"), || (logical "or"), *==* (equals), *!=* (not equals), *>*, *<*, *>=*, *<=* operands.

**Comment by sidoruka** [comment]
Backport to `release/0.16`

**Comment by rodichenko** [comment]
@sidoruka @NShaforostov implemented, backported to `release/0.16` (e5c069562d5775a4e9f13501775c73e5c3e39fe1)

**Comment by NShaforostov** [comment]
Docs were added via #3916 and located [here](https://github.com/epam/cloud-pipeline/blob/develop/docs/md/manual/06_Manage_Pipeline/6.5._Pipeline_configuration.md).
